android/BOINC/app/src/main/java/edu/berkeley/boinc/rpc/ProjectConfigReplyParser.java

```diff
@@ -151,10 +151,6 @@ else if(localName.equalsIgnoreCase("platform") &&
                         platformName = "";
                         platformPlanClass = "";
                     }
-                    else if(localName.equalsIgnoreCase("platform") && withinPlatforms) { // finish platforms
-                        withinPlatforms = false;
-                        mProjectConfig.platforms = mPlatforms;
-                    }
```

Comment:
Don't you have to retain the two statements contained in the then-clause you removed? Shouldn't they be added to original if-clause, say, following line 152?

Reply:
IOW, you could simply have removed the following instead, no?
```
                    }
                    else if(localName.equalsIgnoreCase("platform") && withinPlatforms) { // finish platforms
```

Reply:
No, these lines are located here:
https://github.com/BOINC/boinc/blob/8624ed384ef0e23900b647f33ff5a1a7d7626ae3/android/BOINC/app/src/main/java/edu/berkeley/boinc/rpc/ProjectConfigReplyParser.java#L92-L95

Reply:
Here we're parsing several <platform> tags inside <platforms> tag. If we merge these lines we will have an logic error parsing only one <platform> instead of all,

Reply:
Ok, just wanted to ensure you didn't miss them by accident. Thanks.